src/sagemaker/model.py

```diff
@@ -274,9 +274,17 @@ def _compilation_image_uri(self, region, target_instance_type, framework, framew
             framework (str): The framework name.
             framework_version (str): The framework version.
         """
-        framework_prefix = "inferentia-" if target_instance_type.startswith("ml_inf") else "neo-"
+        framework_prefix = ""
+        framework_suffix = ""
+        
+        if framework == "xgboost":
```

Comment:
i think we need this to include the image_classifiction config as well

Reply:
image-classifiction is not in the allowed framework for neo list. we probably have a bug here. This should be fixed.